presto-main-base/src/main/java/com/facebook/presto/sql/planner/iterative/rule/ExtractSpatialJoins.java

```diff
@@ -467,8 +475,11 @@ else if (alignment < 0) {
                 kdbTree.map(KdbTreeUtils::toJson)));
     }
 
-    private static boolean isSphericalJoin(Metadata metadata, RowExpression firstArgument, RowExpression secondArgument)
+    private static boolean isSphericalJoin(Metadata metadata, RowExpression firstArgument, RowExpression secondArgument, boolean nativeExecution)
     {
+        if (nativeExecution) {
```

Comment:
Please add a comment about why we are doing this. And state that this is a temporary solution and will be removed after spherical geometry types are supported.